### ISSUE TITLE
[BUG] Sync Fail

### ISSUE THREAD

**Issue by 0xlne** [itprojects/InboxPager#26]
Sync initiates, downloaded ~30 mails (out of ~1750), & stopped permanently afterwards.
Each time I press sync now displays the following error:
![image](https://user-images.githubusercontent.com/26036905/65859468-90fb4100-e368-11e9-9923-42f121c10b08.png)

Device: Galaxy Note 10+, Andoid 9/Pie

**Comment by itprojects** [comment]
`length=1; index=2` could be an array out of bounds.

Can you answer the following:
1. Can you use logcat (like in android studio) to get a full log by repeating the sync? (removing your username, pass, and servername)
2. Does the app stop after the error? (Force Close?)
4. What's your free space situation?
5. One idea is to delete this inbox and recreate it. Does that solve the issue?
6. Any other technical info you could provide would be great.

Otherwise the app works with popular services. Even if large inboxes are a recurring issue.

**Comment by itprojects** [comment]
Can you provide more information about the email message, on which it is stuck?

Remove the part of "Subject:" and everything before it. Change the message contents words with "bla bla", instead of whatever somebody communicated to you.

For example: 
This is a private email message -> Bla bla bla bla bla bla.

Leave in the parts with "--od32IJDAHDgdyudgue" and "Content-".

**Comment by 0xlne** [comment]
>     1. Can you use logcat (like in android studio) to get a full log by repeating the sync? (removing your username, pass, and servername)
Uploaded here:
https://bin.disroot.org/?cb79ed0ee1f01e22#C0RlYR1QVAuivfKDRmpz0pDFlwdxlGJO2H3ceGt6Dt0=

>     2. Does the app stop after the error? (Force Close?)
No

>     3. What's your free space situation?
Storage: 206Gb free
RAM: 4.8Gb

>     4. One idea is to delete this inbox and recreate it. Does that solve the issue?
I did, several times. I also just tried installing on another device (Sony Xperia XZ2/Android 9/pie), with the same results. This might indicate an issue with a particular email, maybe Inbox Pager has an issue with exotic/non-alphanumeric characters in the subject line? I unfortunately cannot pinpoint the specific message causing this issue.

>     5. Any other technical info you could provide would be great.
My provider is gmail.
Inbox pager starts refreshing, displays: Fetching message 1/17xx, then displays the error I shared earlier.
Would you be able to prepare a version of IP with more verbose output, especially during syncing? If this is an issue with the subject line, this could help us pinpoint the root cause.

**Comment by 0xlne** [comment]
I cleaned up my inbox a bit & noticed indeed having some mails with non-alphanumeric chars like "@" or "/". Inbox Pager started syncing a few mails again, but stopped, probably on reaching another such message.

**Comment by itprojects** [comment]
>  some mails with non-alphanumeric chars like "@" or "/"

That shouldn't be the issue.

Looking at line 1063 in IMAP this may be a different format issue.

Email messages can be sent in non-text and non-html formats, in addition to them. For example emails can be sent in `.rtf` and other formats. To handle those, InboxPager reads them letter-by-letter, without converting them to readable text. This was implemented, so that the user having received a message, will know that they have a non-empty message, and then, they will go online to see the full message. Not all of your emails have such parts. One way to check is to "see original"  in gmail, starting with the oldest messages. Look for "Content-Type: text/rtf"  but not "Content-Type: text/plain" or "Content-Type: text/html". Report results.

A test version of InboxPager will be uploaded soon, to see if removing supports for this is a fix.

In test the new apk.

You may need to remove the repository InboxPages from your device, before you install the custom made InboxPager apk. This is for security, because android wants apps from a repository not to be upgraded without cryptographic signature of the repository maintainer. If the fixed apk works, when the changes (if they are the solution) will be pushed to the repository. Then your custom InboxPager apk will have to be deleted, before the fixed-repository version can be installed later, for the same reasons.

**Comment by itprojects** [comment]
Try this [apk](https://github.com/itprojects/InboxPager/blob/f23c22b4659ee87d152fb5e3f38dcbca212eba0e/InboxPager-411.apk).

**Comment by 0xlne** [comment]
I pruned my inbox to 872 items now.
Using the new version you shared allowed me to sync around 30 mails, but the same issue appears afterwards:
![image](https://user-images.githubusercontent.com/26036905/66162481-af925e00-e62e-11e9-95ea-7e0006a9ba7f.png)

**Comment by itprojects** [comment]
It might not be the same issue.
Please don't reduce the number of emails, for now.
We need it for testing.

**Comment by itprojects** [comment]
Try it with this [update](https://github.com/itprojects/InboxPager/blob/apk-direct/InboxPager-412.apk).

**Comment by 0xlne** [comment]
It still didn't help, but I did the following:
I assumed that Inbox Pager is syncing starting from the oldest message & I managed to identify the first message which was blocking the sync. I moved it to trash, resumed syncing, lo & behold - sync resumed. Currently 200 out of 872 & still going.
I can forward you the "blocking" mail for analysis if you share your mail address.

**Comment by itprojects** [comment]
When you have a line in the subject like this one:

`UTF-8?Q?LKJSKJWKNHSUJW`

That means that the message subject was sent in an encoding that the email client has to first split and convert the lets say UTF-8.

In simpler terms you were right from the start. But, we may have solved the wrong bug.

Keep the offensive emails in an Imap folder different from inbox. We may yet need them.

Can you offer a screenshot like the ones before, with all the lines leading up to the error?

**Comment by itprojects** [comment]
If you try this [debug](https://github.com/itprojects/InboxPager/blob/apk-direct/InboxPager-413.apk) version, we should confirm the hypothesis.

**Comment by 0xlne** [comment]
![image](https://user-images.githubusercontent.com/26036905/66172677-f7c07900-e64c-11e9-97fa-139f76e014cc.png)

**Comment by itprojects** [comment]
With [this build](https://github.com/itprojects/InboxPager/blob/apk-direct/InboxPager-414.apk), when InboxPager cannot correctly process the attachments it will try to mitigate.

**Comment by 0xlne** [comment]
Thank you - I've been able to sync the rest of my inbox.
The error log contained references to 4 mails with a subject/sender context in UTF, but the app crashed when I tried deleting a few mails.
Upon restart the error log was empty thus I can no longer share that info. Unless I recreate my account in Inbox Pager & sync from scratch.

**Comment by itprojects** [comment]
In the end, how many emails do you have currently?
If you could re-create from scratch and sync just once more, that would be great.
Other than that, it seems there were more than one bugs.
If the fix is confirmed, then we can close this issue?

**Comment by 0xlne** [comment]
I've uninstalled & reinstalled IP with v4.14 you offerred up.
The inbox synced fully - the email count in IP is the same I have when logging into gmail via a browser or k9.
Here's what was present in the error log (edited out a name):
![image](https://user-images.githubusercontent.com/26036905/66252291-0f395800-e75a-11e9-89e7-aebe90811b2a.png)

Taking into consideration the above, I guess we can consider this particular bug, fixed & issue to be closed.

**Comment by itprojects** [comment]
We can safely ignore `!!7` it's the attachment bug, `!!3` is the Quoted-printable.
